### DISCUSSION TITLE
add 76 styles and 7 word libraries(添加76个风格和7个常用词库)

### DISCUSSION THREAD

**Discussion by chouhai2018** [lllyasviel/Fooocus#2885] [Ideas]
声明:
	1.我不是专业的编程人员,有很多不足,请谅解.
	2.内容来自网络,都是找相关资料收集到的,原始出处是哪里,我也不知道.
	3.我只是整合和调整了资料
	4.感谢各位资料原始作者
	5.感谢Fooocus各位,带来这么好的软件
	6.这个文档我也只能写这样了,好看的那种,不会写.
	
安装方法
	下载压缩包 [Fooocus.zip](https://github.com/lllyasviel/Fooocus/files/15249828/Fooocus.zip)
	解压缩文件,	复制Fooocus到你的Fooocus目录就行了
	重新启动Fooocus
	
说明:
	新增wildcards 7个常用词库
	新增sdxl_styles 76个风格

使用说明
	__movements__,__mediums__,__flavors__,by __artists__, 词按照这样的顺序写,会得到很好的图(仅我的测试结果)
	要是提取文件出错,请修改modules/sdxl_styles.py 78行,把那个encoding='utf-8'去掉,我用的中文windows,新建文件不是utf-8的,怎么改都不行,只能改源码了
	修改完成的	words = open(os.path.join(modules.config.path_wildcards, matches[0])).read().splitlines()
	
	风格使用要是没有生效,请仅使用一个风格应用
	
Declaration:

I am not a professional programmer, there are many shortcomings, please understand.
The content comes from the internet, collected from related materials. I don't know the original source.
I simply integrate and adjust the information.
Thanks to all the original authors of the materials.
Thanks to all the folks at Fooocus for bringing such great software.
This document is the best I can write, in a visually appealing way. I'm not good at writing.

Installation Method:
download zip file,Unzip the file, copy Fooocus to your Fooocus directory, and restart Fooocus.
[Fooocus.zip](https://github.com/lllyasviel/Fooocus/files/15249828/Fooocus.zip)
Explanation:
Added 7 common word libraries for wildcards.
Added 76 styles for sdxl_styles.

Instructions for Use:
Write the words in the order of movements,mediums,flavors,by artists to get a good graph (only based on my test results).
If file extraction fails, please modify line 78 in modules/sdxl_styles.py and remove encoding='utf-8'. I use Chinese Windows, and newly created files are not utf-8, no matter how I modify them, it doesn't work. I can only modify the source code.
After modification, words = open(os.path.join(modules.config.path_wildcards, matches[0])).read().splitlines().

If the style application doesn't take effect, please only apply one style.

--------------------------------
language en
	"FF base": "FF base",
	"FF 3D Model": "FF 3D Model",
	"FF Analog Film": "FF Analog Film",
	"FF Anime": "FF Anime",
	"FF Cinematic": "FF Cinematic",
	"FF Comic Book": "FF Comic Book",
	"FF Craft Clay": "FF Craft Clay",
	"FF Digital Art": "FF Digital Art",
	"FF Enhance": "FF Enhance",
	"FF Fantasy Art": "FF Fantasy Art",
	"FF Isometric Style": "FF Isometric Style",
	"FF Line Art": "FF Line Art",
	"FF Lowpoly": "FF Lowpoly",
	"FF Neon Punk": "FF Neon Punk",
	"FF Origami": "FF Origami",
	"FF Photographic": "FF Photographic",
	"FF Pixel Art": "FF Pixel Art",
	"FF Texture": "FF Texture",
	"FF Advertising": "FF Advertising",
	"FF Food Photography": "FF Food Photography",
	"FF Real Estate": "FF Real Estate",
	"FF Abstract": "FF Abstract",
	"FF Cubist": "FF Cubist",
	"FF Graffiti": "FF Graffiti",
	"FF Hyperrealism": "FF Hyperrealism",
	"FF Impressionist": "FF Impressionist",
	"FF Pointillism": "FF Pointillism",
	"FF Pop Art": "FF Pop Art",
	"FF Psychedelic": "FF Psychedelic",
	"FF Renaissance": "FF Renaissance",
	"FF Steampunk": "FF Steampunk",
	"FF Surrealist": "FF Surrealist",
	"FF Typography": "FF Typography",
	"FF Watercolor": "FF Watercolor",
	"FF Fighting Game": "FF Fighting Game",
	"FF GTA": "FF GTA",
	"FF Super Mario": "FF Super Mario",
	"FF Minecraft": "FF Minecraft",
	"FF Pokémon": "FF Pokémon",
	"FF Retro Arcade": "FF Retro Arcade",
	"FF Retro Game": "FF Retro Game",
	"FF RPG Fantasy Game": "FF RPG Fantasy Game",
	"FF Strategy Game": "FF Strategy Game",
	"FF Street Fighter": "FF Street Fighter",
	"FF Legend of Zelda": "FF Legend of Zelda",
	"FF Architectural": "FF Architectural",
	"FF Disco": "FF Disco",
	"FF Dreamscape": "FF Dreamscape",
	"FF Dystopian": "FF Dystopian",
	"FF Fairy Tale": "FF Fairy Tale",
	"FF Gothic": "FF Gothic",
	"FF Grunge": "FF Grunge",
	"FF Horror": "FF Horror",
	"FF Minimalist": "FF Minimalist",
	"FF Monochrome": "FF Monochrome",
	"FF Nautical": "FF Nautical",
	"FF Space": "FF Space",
	"FF Stained Glass": "FF Stained Glass",
	"FF Techwear Fashion": "FF Techwear Fashion",
	"FF Tribal": "FF Tribal",
	"FF Zentangle": "FF Zentangle",
	"FF Collage": "FF Collage",
	"FF Flat Papercut": "FF Flat Papercut",
	"FF Kirigami": "FF Kirigami",
	"FF Paper Mache": "FF Paper Mache",
	"FF Paper Quilling": "FF Paper Quilling",
	"FF Papercut Collage": "FF Papercut Collage",
	"FF Papercut Shadow Box": "FF Papercut Shadow Box",
	"FF Stacked Papercut": "FF Stacked Papercut",
	"FF Thick Layered Papercut": "FF Thick Layered Papercut",
	"FF Alien": "FF Alien",
	"FF Film Noir": "FF Film Noir",
	"FF HDR": "FF HDR",
	"FF Long Exposure": "FF Long Exposure",
	"FF Neon Noir": "FF Neon Noir",
	"FF Silhouette": "FF Silhouette",
	"FF Tilt Shift": "FF Tilt Shift",
	
中文汉化
	"Fei 3D Model": "Fei 3D模型",
	"Fei Analog Film": "Fei 模拟胶片",
	"Fei Anime": "Fei 动漫",
	"Fei Cinematic": "Fei 电影风格",
	"Fei Comic Book": "Fei 漫画书",
	"Fei Craft Clay": "Fei 手工黏土",
	"Fei Digital Art": "Fei 数字艺术",
	"Fei Enhance": "Fei 增强",
	"Fei Fantasy Art": "Fei 幻想艺术",
	"Fei Isometric Style": "Fei 等距风格",
	"Fei Line Art": "Fei 线条艺术",
	"Fei Lowpoly": "Fei 低多边形",
	"Fei Neon Punk": "Fei 霓虹庞克",
	"Fei Origami": "Fei 折纸艺术",
	"Fei Photographic": "Fei 摄影",
	"Fei Pixel Art": "Fei 像素艺术",
	"Fei Texture": "Fei 纹理",
	"Fei Advertising": "Fei 广告",
	"Fei Food Photography": "Fei 美食摄影",
	"Fei Real Estate": "Fei 房地产",
	"Fei Abstract": "Fei 抽象艺术",
	"Fei Cubist": "Fei 立体主义",
	"Fei Graffiti": "Fei 涂鸦",
	"Fei Hyperrealism": "Fei 超现实主义",
	"Fei Impressionist": "Fei 印象派",
	"Fei Pointillism": "Fei 点彩派",
	"Fei Pop Art": "Fei 波普艺术",
	"Fei Psychedelic": "Fei 迷幻艺术",
	"Fei Renaissance": "Fei 文艺复兴",
	"Fei Steampunk": "Fei 蒸汽朋克",
	"Fei Surrealist": "Fei 超现实主义",
	"Fei Typography": "Fei 排版艺术",
	"Fei Watercolor": "Fei 水彩艺术",
	"Fei Fighting Game": "Fei 格斗游戏",
	"Fei GTA": "Fei 侠盗猎车手",
	"Fei Super Mario": "Fei 超级马里奥",
	"Fei Minecraft": "Fei 我的世界",
	"Fei Pokémon": "Fei 宝可梦",
	"Fei Retro Arcade": "Fei 复古街机",
	"Fei Retro Game": "Fei 复古游戏",
	"Fei RPG Fantasy Game": "Fei 奇幻角色扮演游戏",
	"Fei Strategy Game": "Fei 策略游戏",
	"Fei Street Fighter": "Fei 街头霸王",
	"Fei Legend of Zelda": "Fei 塞尔达传说",
	"Fei Architectural": "Fei 建筑设计",
	"Fei Disco": "Fei 迪斯科",
	"Fei Dreamscape": "Fei 梦境风景",
	"Fei Dystopian": "Fei 反乌托邦",
	"Fei Fairy Tale": "Fei 童话故事",
	"Fei Gothic": "Fei 哥特风格",
	"Fei Grunge": "Fei 垮掉风",
	"Fei Horror": "Fei 恐怖",
	"Fei Minimalist": "Fei 极简主义",
	"Fei Monochrome": "Fei 单色",
	"Fei Nautical": "Fei 航海",
	"Fei Space": "Fei 太空",
	"Fei Stained Glass": "Fei 彩色玻璃",
	"Fei Techwear Fashion": "Fei 科技穿戴",
	"Fei Tribal": "Fei 部落风",
	"Fei Zentangle": "Fei 禅绕",
	"Fei Collage": "Fei 拼贴",
	"Fei Flat Papercut": "Fei 扁平剪纸",
	"Fei Kirigami": "Fei 切纸艺术",
	"Fei Paper Mache": "Fei 纸浆艺术",
	"Fei Paper Quilling": "Fei 纸条艺术",
	"Fei Papercut Collage": "Fei 剪纸拼贴",
	"Fei Papercut Shadow Box": "Fei 剪纸影箱",
	"Fei Stacked Papercut": "Fei 叠加剪纸",
	"Fei Thick Layered Papercut": "Fei 厚层剪纸",
	"Fei Alien": "Fei 外星人",
	"Fei Film Noir": "Fei 黑色电影",
	"Fei HDR": "Fei 高动态范围",
	"Fei Long Exposure": "Fei 长时间曝光",
	"Fei Neon Noir": "Fei 霓虹黑",
	"Fei Silhouette": "Fei 轮廓剪影",
	"Fei Tilt Shift": "Fei 倾斜移轴",

**Comment by xhoxye** [comment]
最好不要修改原代码，因为更新会覆盖，创建和转换文本文件到UTF-8编码可以使用文本编辑器如editplus，VSCode, notepad
It is best not to modify the original code, as the update will overwrite, create and convert the text file to UTF-8 encoding using a text editor such as editplus, VSCode, notepad
![QQ截图20240509090048](https://github.com/lllyasviel/Fooocus/assets/129571231/4d4d7dfd-f1cd-445a-8c6b-6d4b4e872bc8)
![QQ截图20240509090504](https://github.com/lllyasviel/Fooocus/assets/129571231/5ca42787-bc00-41ab-94a2-75a9e7a46393)
![QQ截图20240509090454](https://github.com/lllyasviel/Fooocus/assets/129571231/613afea5-a671-42f3-b938-5d57d208c7c0)
